orttraining/orttraining/training_api/onnxruntime_training_c_api.cc

```diff
@@ -210,16 +216,34 @@ ORT_API_STATUS_IMPL(OrtTrainingApis::OptimizerStep, _Inout_ OrtTrainingSession*
   API_IMPL_END
 }
 
-ORT_API_STATUS_IMPL(OrtTrainingApis::RegisterLinearLRScheduler, _Inout_ OrtTrainingSession* sess,
-                    _In_ int64_t warmup_step_count, _In_ int64_t total_step_count) {
+ORT_API_STATUS_IMPL(OrtTrainingApis::RegisterLRScheduler, _Inout_ OrtTrainingSession* sess,
+                    _In_ void* lr_scheduler_parameters, _In_ enum OrtLRSchedulerType lr_scheduler_type,
+                    _In_opt_ const float* initial_lr) {
   API_IMPL_BEGIN
 
+  OrtStatus* status = nullptr;
+
   auto session = reinterpret_cast<onnxruntime::training::api::TrainingSession*>(sess);
-  ORT_API_RETURN_IF_STATUS_NOT_OK(session->RegisterScheduler([warmup_step_count, total_step_count](auto optimizer) {
-    return std::make_unique<onnxruntime::training::api::LinearLRScheduler>(optimizer, warmup_step_count, total_step_count);
-  }));
+  switch (lr_scheduler_type) {
+    case OrtLRSchedulerType::LinearLRScheduler: {
+      auto parameters = reinterpret_cast<OrtLinearLRSchedulerParameters*>(lr_scheduler_parameters);
+      ORT_API_RETURN_IF_STATUS_NOT_OK(session->RegisterScheduler([&parameters](auto optimizer) {
+        return std::make_unique<onnxruntime::training::api::LinearLRScheduler>(
+            optimizer, parameters->warmup_step_count, parameters->total_step_count);
+      }));
+
+      if (initial_lr) {
```

Comment:
For our API, if a nullptr initial_lr is given, should give an explictily error message?

Reply:
Oh I see, it is an optional param.